### DISCUSSION TITLE
my opendeck-64-rotary-controller

### DISCUSSION THREAD

**Discussion by reelex** [shanteacontrols/OpenDeck#159] [Show and tell]
Hi,

I want to show you my little midi controller that I realised with the help of open deck.
I was in urgent need of a midicontroller to play my VST-heavy music live and I just couldn't afford most of the stuff on the market and was also missing controllers with a higher resolution than 7bit. OpenDeck was the perfect solution for me and I will probably create a lot more Midi-Controllers with this method. So for this project I wanted as many rotary knobs with a high resolution on a build that is also as small as possible. And this is what I ended up with:

![photo_2022-06-18_17-18-06](https://user-images.githubusercontent.com/104325628/174446278-55ee9510-db74-4e2e-90d3-a2d07afd79bf.jpg)

![photo_2022-06-18_17-18-17](https://user-images.githubusercontent.com/104325628/174446282-7fd4774f-3677-4387-946c-3b1401aa3909.jpg)

It has 64 knobs, most of them working as 14bit-ableton-midi but I couldn't use it for all of them so a few also send pitchbend-midi.
I designed it fully digitally and I am very happy with the design. I would only alter the screw-connections that hold the case together. I would use less screws, but make the connections stronger. Because of the limitation of my 3d-printer I had to split the case in half. If I had a bigger 3d-printer I would of course also print it all in one, which would also make it more sturdy. I am super happy with the surface finish though. Some testing with the print parameters and printing it on a smooth glass plate were really worth it. 

![2](https://user-images.githubusercontent.com/104325628/174446519-aedd4271-26be-49d8-809c-1c0b3b3d3471.jpg)
![3](https://user-images.githubusercontent.com/104325628/174446526-0905a6ad-14ca-4bae-a4c7-42c476ec6f9f.jpg)
![photo_2022-06-18_17-25-04](https://user-images.githubusercontent.com/104325628/174446554-33c286c7-b5fd-4826-82a0-00293f9e9cd4.jpg)
![photo_2022-06-18_17-25-07](https://user-images.githubusercontent.com/104325628/174446564-f46d3be3-d249-4efe-ae7c-2020c82788a7.jpg)
![photo_2022-06-18_17-24-58](https://user-images.githubusercontent.com/104325628/174446562-a956def1-2f35-428e-92b1-5b0f27cde4da.jpg)
![photo_2022-06-18_18-24-10](https://user-images.githubusercontent.com/104325628/174447725-a4b79f20-cb35-4d35-97d9-5787881db147.jpg)

I also experimented with different caps as most of the ones you can buy look ugly or don't fit the 3d-print look and are also way to big for my very tight design. I found a design that has good grip, a good size and also looks funny and interesting. The different colors help to find your way while playing and not mix up the knobs. Of course I can change the pattern depending on my assignments.

![4](https://user-images.githubusercontent.com/104325628/174446637-c8082a25-ce5d-4655-9b2e-fd1fe3173161.jpg)
![photo_2022-06-18_17-25-01](https://user-images.githubusercontent.com/104325628/174446643-fcea359d-346d-465e-b446-cbb51c4fd259.jpg)

It works like a charm and it turned out to have just the right size for my fingers and compared to my laptop.
![photo_2022-06-18_17-42-29](https://user-images.githubusercontent.com/104325628/174446784-7ad24dd7-f023-40f6-b129-1c4b89dabf86.jpg)

My next plan is to use some of the testprints I have left to give my 64-rotary-deck a smaller brother with 32 buttons. :)
![photo_2022-06-18_17-18-28](https://user-images.githubusercontent.com/104325628/174446830-80450f50-64ca-4581-a86e-9159a9467436.jpg)

Thanks again paradajz!
I have a gig in July where I will use the controller.
I am already excited. :D

**Comment by paradajz** [comment]
Doesn't look so little to me! It's a great design. I'm glad my work made it possible.

Are you satisfied with the resolution you're getting out of the firmware? The 14 bit values are derived from 12 bit ADC, so the values are scaled.

Any reason why you didn't use PCBs for pots? Would save you a lot of wires. :) You can find examples of those in bin/sch directory in repository.

**Reply by reelex** [reply]
Well 12 bit is already a LOT better than 7 bit, so I'm very happy and the scaling seems to work as intended. I was not put of by the scaling while using it, so it works fine. Of course I'd wish microcontrollers would have 14 bit ADCs... that would be awesome. But there are not many 14bit midi controllers anyways and It's such a good idea from you to just scale it to send 14bit midi. That way it gives you at least the option to use the full 12bit given by the hardware. I was very happy to see that! Pragmatic approach with a big impact for me. :D

Well maybe next time I'll use PCBs... ^^ Never designed one myself though. Also I wanted to save space as I didn't leave much space at the bottom and I also wanted a very sleek design and well, cables bend... PCBs don't. But soldering was a mess, so I'll definitely look into it. Thanks for the suggestion!

**Reply by paradajz** [reply]
Thanks for the feedback! Mind if I put the pictures of your controller on the gallery on my site?

**Reply by reelex** [reply]
Very welcome! Yeah of course, take the pics! :)

**Comment by bigggmoustache** [comment]
Is this going to be open source eventually?

**Reply by reelex** [reply]
Hi, you mean the 3d-model? I can upload it, for sure. I would love other people to use it. It still has some issues but if you know CAD, you can solve them. :) I use rhino and i can share the rhino-files. I dont know how to export the model for other CAD software though. And also, the file is REEEEAALY messy because I made it in quite a hurry. Lots of unnamed layers.^^ I could also just share a ready to 3d-print 3mf-file. But of course that is not real 'open source' as it's hard to change anything.

**Comment by bigggmoustache** [comment]
I mean the whole project tbh lol. I've been searching for a diy 60 knob
alternative for a long time now and I gotta say you knock it out of the
park bud.
